### PR TITLE
Make the installation of Windows unattended

### PR DESCRIPTION
**Pull request by pablo-herranz**
By adding an `Autounattend.xml` file to the ISO root folder with all the options defined, the installation of Windows runs unattended without need of any clicks or screens.

- Related ticket: https://progress.opensuse.org/issues/121033
- Needles:
- Verification runs:
  - [Windows 10 BIOS](https://openqa.suse.de/tests/12373524)
  - [Windows 10 UEFI](https://openqa.suse.de/tests/12373672)
  - [Windows 11 UEFI](https://openqa.suse.de/tests/12374123)

  - [Job running on a PUBLISH_HDD_1](https://openqa.suse.de/tests/12374209)
  - [Win10 job group after changes](https://openqa.suse.de/tests/overview?distri=windows&version=10&build=22H2&groupid=287)
  - [Win11 job group after changes](https://openqa.suse.de/tests/overview?distri=windows&version=11&build=22H2&groupid=287)

*NOTE: the keys used in the XML files are the default ones provided by Microsoft for installing without activation*

### CODE REVIEW

File: data/wsl/create_unattend_iso.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+fi
+
+# if [$2 -ne "BIOS"]||[$2 -ne "UEFI"]; then
+#     echo "You should declare UEFI or BIOS!"
+#     echo "  USAGE: ${0} <ISO_FILE> <BIOS/UEFI>"
+#     exit
+# fi
+
+basename=`echo $1 | cut -d '.' -f 1`
+echo "***** Creating ${basename}_${2}_unattend.iso from ${1} *****"
```
Comment (mloviska) [inline data/wsl/create_unattend_iso.sh:16]: what should be the second parameter then?

**Comment by mloviska** [comment]
I am not entirely sure whether it is a good idea to remove the attended installation process as we are changing a supported way how to install windows for unsupported. I would rather keep the old code, as it might be useful.

File: data/wsl/create_unattend_iso.sh (hunk 1 above)
Comment (pablo-herranz) [reply data/wsl/create_unattend_iso.sh]: The second parameter is supposed to be 'UEFI' or 'BIOS' string. Anyway, I removed the bash script

**Comment by pablo-herranz** [comment]
> I am not entirely sure whether it is a good idea to remove the attended installation process as we are changing a supported way how to install windows for unsupported. I would rather keep the old code, as it might be useful.

Ok, I remade the code so that there's a new variable called `WIN_ATTENDED`, which mades the test be unattended or not. Also I created [this MR](https://gitlab.suse.de/qac/qac-openqa-yaml/-/merge_requests/1346) to add new tests in the job group, as the attended/unattended depends on the ISO too.

File: data/wsl/create_unattend_iso.sh (hunk 1 above)
Comment (mloviska) [reply data/wsl/create_unattend_iso.sh]: Pity, I found it useful and as good idea

Comment (pablo-herranz) [reply data/wsl/create_unattend_iso.sh]: I'll add this script to our Confluence page, and maybe in the future we can create a job to auto generate the ISOs

File: tests/wsl/install/ms_win_installation.pm [hunk 2]
```diff
@@ -26,53 +26,63 @@ sub prepare_win11 {
...
+        send_key 'alt-c';    # custom
+        assert_screen 'windows-disk-partitioning';
+        send_key 'alt-l';    # load driver
+        assert_screen 'windows-load-driver';
+        send_key 'alt-b';    # browse button
+        send_key 'c';
+        save_screenshot;
+        send_key 'c';    # go to second CD drive with drivers
+        send_key 'right';    # ok
+        sleep 0.5;
```
Comment (grisu48) [inline tests/wsl/install/ms_win_installation.pm:74]: Is this needed? If so, is sleeping for 500ms really enough, even if the workers are under heavy load?

Comment (pablo-herranz) [reply tests/wsl/install/ms_win_installation.pm:74]: That has always been there and in fact the "good practices guide" recommends not to use `sleep` :man_shrugging:

Comment (grisu48) [reply tests/wsl/install/ms_win_installation.pm:74]: I'm not against using `sleep` here, but such a short sleep appears a bit ... short.

But ok, if this was working so far, let's keep it.

**Review by grisu48 — APPROVED** [review]
LGTM, if the VRs are up-to-date and passing.

**Review by ricardobranco777 — APPROVED** [review]

File: data/wsl/Autounattend_BIOS.xml [hunk 3]
```diff
@@ -0,0 +1,363 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (clanig) [inline data/wsl/Autounattend_BIOS.xml:1]: As mentioned elsewhere, there might be some clarifications regarding the workflow.

**Review by clanig — CHANGES_REQUESTED** [review]

**Review by clanig — APPROVED** [review]

**Pull request merged**